test/suites/host-ip-change.robot

```diff
@@ -0,0 +1,233 @@
+  # and configured properly.
+  IF  '${SSH_PRIV_KEY}'
+    SSHLibrary.Login With Public Key  ${USHIFT_USER}  ${SSH_PRIV_KEY}
+  ELSE
+    SSHLibrary.Login  ${USHIFT_USER}  allow_agent=True
+  END
+
+Wait and Retry Login MicroShift Host
+  [Arguments]  ${ip}
+  [Documentation]  Wait and retry logging in to the MicroShift host via SSH
+  ...  with a specified maximum number of retries and interval between retries.
+  Wait Until Keyword Succeeds  20x  15s  SSHLibrary.Login With Public Key  ${USHIFT_USER}  ${SSH_PRIV_KEY}
+
+Reboot Microshift System
+    [Documentation]    Reboot the system using "systemctl reboot" command
+    ${stdout}    ${rc}=    Execute Command    systemctl reboot
+    ...    sudo=True    return_rc=True
+    Log    Reboot command executed. Return code: ${rc}
+    Run Keyword If    ${rc} != 0    Fail    Reboot command failed with return code ${rc}
+
+Verify Microshift Show Config Command Output
+    [Arguments]  ${new_host_ip}
+    [Documentation]   Verify that 'microshift show-config' command output contains the IP address '10.1.235.91'
+    ${stdout}    ${rc}=    Execute Command    microshift show-config
+    ...    sudo=True    return_rc=True
+    Log    Microshift command executed. Return Outpu: ${stdout}
+    Should Contain    ${stdout}    ${new_host_ip}
+    Should Be Equal As Integers    ${rc}    0
+
+Check new Kubeconfig generated for new IP
+    [Documentation]    Check the new kubeconfig
+     ${hostname}   ${rc1}=    Execute Command    hostname
+    ...    sudo=True    return_rc=True
+    ${output}    ${rc2}    Execute Command    ls /var/lib/microshift/resources/kubeadmin/${hostname}/
+    ...    sudo=True    return_rc=True
+    Log    Check Microshift new kubeconfig command executed. Return Output: ${output}
+    Should Contain    ${output}    kubeconfig
+    Should Be Equal As Integers    ${rc1}    0
+    Should Be Equal As Integers    ${rc2}    0
+
+Get File Checksum
+    [Arguments]    ${file_path}
+    ${stdout}    ${rc}=    Execute Command     sha256sum ${file_path}
+    ...    sudo=True    return_rc=True
+    Run Keyword If    '${rc}' != '0'    Fail    Failed to get the checksum of the file: ${file_path}
+    [Return]    ${stdout.split()[0]}
+
+Compare File Checksum
+    [Arguments]    ${file_path}    ${expected_checksum}
+    ${current_checksum}=    Get File Checksum    ${file_path}
+    Should Not Be Equal    ${current_checksum}    ${expected_checksum}
+
+*** Test Cases ***
+Verify certificate updates when the host IP changes 
+    [Documentation]  Ensure certificate updates when the host IP changes
+    
+    # Generate new IP in same subnet
+    ${result}=    Generate New IP
+    # Assign the values to global variables
+    Set Global Variable    ${new_host_ip}    ${result}[0]
+    Set Global Variable    ${gateway}       ${result}[1]
+    Set Global Variable    ${device}        ${result}[2]
+    Set Global Variable    ${nmconnection_path}      /etc/NetworkManager/system-connections/${device}.nmconnection
```

Comment:
Can we set the global variables in the `Generate New IP` function, rather than parsing the stdout?

Reply:
Updated